internal/httpclient/client.go

```
@@ -49,11 +114,16 @@ func newStreamingTransport() *http.Transport {
 		// Icecast and SHOUTcast servers only support HTTP/1.x.
 		config.NextProtos = nil
 
-		conn, err := (&tls.Dialer{Config: config}).DialContext(ctx, network, addr)
+		rawConn, err := rawDial(ctx, network, addr)
 		if err != nil {
 			return nil, fmt.Errorf("dial TLS %s: %w", addr, err)
 		}
-		return newICYConn(conn), nil
+		tlsConn := tls.Client(rawConn, config)
+		if err := tlsConn.HandshakeContext(ctx); err != nil {
+			rawConn.Close()
```

golangci-lint (2.13.2)

[error] Error return value of `rawConn.Close` is not checked

(errcheck)